ddtrace/opentracer/span.py

```diff
@@ -1,39 +1,157 @@
 
     def finish(self, finish_time=None):
-        """"""
-        pass
+        """Finish the span.
+
+        This calls finish on the ddspan.
+        """
+        if self.finished:
+            return
+
+        # finish the datadog span
+        self._span.finish()
+        self.finished = True
+
+    def set_baggage_item(self, key, value):
+        """Sets a baggage item in the span context of this span.
+
+        Baggage is used to propagate state between spans.
 
-    def get_baggage_item(self, key_values, timestamp=None):
-        """"""
-        pass
+        :param key: baggage item key
+        :type key: str
+
+        :param value: baggage item value
+        :type value: a type that can be compat.stringify()'d
+
+        :rtype: Span
+        :return: itself for chaining calls
+        """
+        self._context.set_baggage_item(key, value)
+        return self
+
+    def get_baggage_item(self, key):
+        """Gets a baggage item from the span context of this span.
+
+        :param key: baggage item key
+        :type key: str
+
+        :rtype: str
+        :return: the baggage value for the given key or ``None``.
+        """
+        return self._context.get_baggage_item(key)
 
     def set_operation_name(self, operation_name):
         """Set the operation name."""
-        pass
+        self._span.name = operation_name
 
     def log_kv(self, key_values, timestamp=None):
-        """"""
-        pass
+        """Add a log record to this span.
+
+        Passes on relevant opentracing key values onto the datadog span.
+
+        :param key_values: a dict of string keys and values of any type
+        :type key_values: dict
+
+        :param timestamp: a unix timestamp per time.time()
+        :type timestamp: float
+
+        :return: the span itself, for call chaining
+        :rtype: Span
+        """
 
-    @property
-    def context(self):
-        """"""
-        pass
+        # add the record to the log
+        # TODO: there really isn't any functionality provided in ddtrace
+        #       (or even opentracing) for logging
+        self.log.add_record(key_values, timestamp)
 
-    @property
-    def tracer(self):
-        """"""
-        pass
+        # match opentracing defined keys to datadog functionality
+        # opentracing/specification/blob/1be630515dafd4d2a468d083300900f89f28e24d/semantic_conventions.md#log-fields-table
+        for key in key_values:
+            val = key_values[key]
+            if key == 'event' and val == 'error':
+                # TODO: not sure if it's actually necessary to set the error manually
+                self._span.error = 1
+                self.set_tag('error', 1)
+            elif key == 'error' or key == 'error.object':
+                self.set_tag(errors.ERROR_TYPE, val)
+            elif key == 'message':
+                self.set_tag(errors.ERROR_MSG, val)
+            elif key == 'stack':
+                self.set_tag(errors.ERROR_STACK, val)
```

Comment:
For these last two cases, it seems like they would only ever get called on non-error traces?

Reply:
This just appears to be the supported keys that the datadog tracer supports. Out of the ones listed here:
https://github.com/opentracing/specification/blob/master/semantic_conventions.md#log-fields-table

I mimicked the functionality from our go tracer's opentracing implementation:
https://github.com/DataDog/dd-trace-go/blob/v1/ddtrace/opentracer/span.go#L51-L54